Framework/DataHandling/test/LoadEventNexusIndexSetupTest.h

```diff
@@ -0,0 +1,370 @@
+    TS_ASSERT_EQUALS(indexSetup.eventIDLimits().second, 11);
+    TS_ASSERT_EQUALS(indexInfo.size(), 2);
+    TS_ASSERT_EQUALS(indexInfo.spectrumNumber(0), SpectrumNumber(2));
+    TS_ASSERT_EQUALS(indexInfo.spectrumNumber(1), SpectrumNumber(11));
+    const auto specDefs = indexInfo.spectrumDefinitions();
+    TS_ASSERT_EQUALS(specDefs->at(0), SpectrumDefinition(1));
+    TS_ASSERT_EQUALS(specDefs->at(1), SpectrumDefinition(2));
+  }
+
+  void test_makeIndexInfo_min_range() {
+    LoadEventNexusIndexSetup indexSetup(m_ws, 11, EMPTY_INT(), {1});
+    const auto indexInfo = indexSetup.makeIndexInfo();
+    TS_ASSERT_EQUALS(indexSetup.eventIDLimits().first, 1);
+    TS_ASSERT_EQUALS(indexSetup.eventIDLimits().second, 12);
+    TS_ASSERT_EQUALS(indexInfo.size(), 3);
+    TS_ASSERT_EQUALS(indexInfo.spectrumNumber(0), SpectrumNumber(1));
+    TS_ASSERT_EQUALS(indexInfo.spectrumNumber(1), SpectrumNumber(11));
+    TS_ASSERT_EQUALS(indexInfo.spectrumNumber(2), SpectrumNumber(12));
+    const auto specDefs = indexInfo.spectrumDefinitions();
+    TS_ASSERT_EQUALS(specDefs->at(0), SpectrumDefinition(0));
+    TS_ASSERT_EQUALS(specDefs->at(1), SpectrumDefinition(2));
+    TS_ASSERT_EQUALS(specDefs->at(2), SpectrumDefinition(3));
+  }
+
+  void test_makeIndexInfo_min_max_range() {
+    LoadEventNexusIndexSetup indexSetup(m_ws, 2, 11, {1});
+    const auto indexInfo = indexSetup.makeIndexInfo();
+    TS_ASSERT_EQUALS(indexSetup.eventIDLimits().first, 1);
+    TS_ASSERT_EQUALS(indexSetup.eventIDLimits().second, 11);
+    TS_ASSERT_EQUALS(indexInfo.size(), 3);
+    TS_ASSERT_EQUALS(indexInfo.spectrumNumber(0), SpectrumNumber(1));
+    TS_ASSERT_EQUALS(indexInfo.spectrumNumber(1), SpectrumNumber(2));
+    TS_ASSERT_EQUALS(indexInfo.spectrumNumber(2), SpectrumNumber(11));
+    const auto specDefs = indexInfo.spectrumDefinitions();
+    TS_ASSERT_EQUALS(specDefs->at(0), SpectrumDefinition(0));
+    TS_ASSERT_EQUALS(specDefs->at(1), SpectrumDefinition(1));
+    TS_ASSERT_EQUALS(specDefs->at(2), SpectrumDefinition(2));
+  }
+
+  void test_makeIndexInfo_from_bank() {
+    LoadEventNexusIndexSetup indexSetup(m_ws, EMPTY_INT(), EMPTY_INT(), {});
+    const auto indexInfo = indexSetup.makeIndexInfo({"det-2", "det-12"});
+    TS_ASSERT_EQUALS(indexSetup.eventIDLimits().first, EMPTY_INT());
+    TS_ASSERT_EQUALS(indexSetup.eventIDLimits().second, EMPTY_INT());
+    TS_ASSERT_EQUALS(indexInfo.size(), 2);
+    // Note that spectrum numbers are simply starting at 1, i.e., are not the
+    // same as without bank-filtering. This is consistent with the old behavior.
+    TS_ASSERT_EQUALS(indexInfo.spectrumNumber(0), SpectrumNumber(1));
+    TS_ASSERT_EQUALS(indexInfo.spectrumNumber(1), SpectrumNumber(2));
+    const auto specDefs = indexInfo.spectrumDefinitions();
+    TS_ASSERT_EQUALS(specDefs->at(0), SpectrumDefinition(1));
+    TS_ASSERT_EQUALS(specDefs->at(1), SpectrumDefinition(3));
+  }
+
+  void test_makeIndexInfo_from_bank_filter_ignored() {
```

Comment:
Could you group these legacy/characterisation tests separate from the tests that exercise desired behaviour. I think that would be very useful for future refactoring work. Naming the tests as such and putting them in a block together would work I think.

Reply:
Is there any apart from this that does not also check desired behavior?